### PR TITLE
chore: add regression tests for server-side-rendering

### PR DESCRIPTION
**Pull request by NamNg1121**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -99,3 +99,31 @@ jobs:
           yarn serve &
       - name: Run regression tests
         run: npx testcafe ${{ matrix.browser }}:headless ./src/test.ts
+  ssh-regression-next:
```
Comment (Brailor) [inline .github/workflows/release.yml:102]: `ssh` -> `ssr`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -99,3 +99,31 @@ jobs:
...
+    needs: release
+      if: "!startsWith(github.event.head_commit.message, 'chore(release)')"
+      name: Regression testing
```
Comment (Brailor) [inline .github/workflows/release.yml:105]: Would be nice to give it a different name than the previous job.

File: regression-test/src/pages/_app.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import Head from 'next/head'
+import type { AppProps } from 'next/app'
+import { InstUISettingsProvider } from '@instructure/emotion'
+import { Alert, View } from '@instructure/ui'
```
Comment (Brailor) [inline regression-test/src/pages/_app.tsx:4]: `Alert`, `View`, and the `* as Components` imports are unused, please delete them.

File: regression-test/package.json [hunk 4]
```diff
@@ -6,6 +6,8 @@
   "scripts": {
     "build:webpack4": "webpack",
     "build:esbuild": "node esbuild.cjs",
+    "build:next": "next build ./src",
+    "start:next": "next start ./src ",
```
Comment (Brailor) [inline regression-test/package.json:10]: This has to be changed to `next dev ./src`, since we wan't to test the development build of the next.js app.

File: regression-test/cypress/e2e/spec.cy.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  })
+
+  it('open breadcrumb subpage', () => {
+    cy.visit('http://localhost:3000')
+    cy.get('a').contains('breadcrumb').click()
+    cy.get('body').should('contain', 'Rabbit Is Rich')
+    cy.get('body').contains('The Rabbit Novels').click()
+    const redirect = 'https://instructure.design'
+    cy.origin(redirect, { args: { redirect } }, ({ redirect }) => {
+      cy.log(redirect)
```
Comment (git-nandor) [inline regression-test/cypress/e2e/spec.cy.ts:23]: Is this just for debugging, or does it have a reason?

File: regression-test/src/app/components/MODAL_IUI.js [hunk 6]
```diff
@@ -26,6 +26,7 @@ import { Modal } from '@instructure/ui'
 import React from 'react'
 const MODAL_IUI = () => {
   return (
+    // <></>
```
Comment (git-nandor) [inline regression-test/src/app/components/MODAL_IUI.js:29]: Maybe this was just left here by accident.

**Review by git-nandor — APPROVED** [review]

File: regression-test/README.md [hunk 7]
```diff
@@ -4,18 +4,10 @@ This app is a target against our regression testing workflow. It simply renders
 
 The regression testing suite is fairly simple at this point, it will only check a couple of things:
 
-- use the latest snapshot version of instui
-- check whether every component is in a renderable state (simply render them)
-- check whether any error/warning is thrown inside the application
+- use the latest snapshot version of InstUI
+- check components are in a renderable state (simply render them)
```
Comment (balzss) [inline regression-test/README.md:8]: minor typo: check *if*

File: regression-test/README.md [hunk 8]
```diff
@@ -24,18 +16,18 @@ In order to run the test you have to have it's dependencies installed. (this pro
...
+Then build the project:
 
 ```sh
 # cwd: regression-test
-npm run build:esbuild
+npm run build
 ```
 
-The regression testing tool (`testcafe`) will test the app in a real browser, so we have to have a server serving our application:
+The regression testing tool will test the app in a real browser, so we have to have a server serving our application:
```
Comment (balzss) [inline regression-test/README.md:26]: I would reword this saying it is starting the next server, also you don't need `run` before `start` so it's just `npm start`

File: regression-test/src/app/components/[componentName]/page.tsx [hunk 9]
```diff
@@ -21,15 +21,22 @@
...
-
+'use client'
 import React from 'react'
-const APPLYTEXTDIRECTION_IUI = () => {
+import { View } from '@instructure/ui'
+
+export default function Component({ params }: any) {
+  const Component = require(`../../components/${params.componentName}`)
+    .default as React.ElementType
   return (
```
Comment (balzss) [inline regression-test/src/app/components/[componentName]/page.tsx:31]: should we wrap this in InstuiSettingsProvider to more accurately simulate realworld usage?

**Review by balzss — APPROVED** [review]
left a couple comments but overall looks great, nice work

File: regression-test/src/app/components/MODAL_IUI.js (hunk 6 above)
Comment (joyenjoyer) [reply regression-test/src/app/components/MODAL_IUI.js:29]: I removed it, thanks.

File: regression-test/README.md (hunk 7 above)
Comment (joyenjoyer) [reply regression-test/README.md:8]: Thanks.

File: regression-test/README.md (hunk 8 above)
Comment (joyenjoyer) [reply regression-test/README.md:26]: Good point.

File: regression-test/src/app/components/[componentName]/page.tsx (hunk 9 above)
Comment (joyenjoyer) [reply regression-test/src/app/components/[componentName]/page.tsx:31]: I added it.
